sql/select.go

```diff
@@ -112,6 +121,7 @@ func (p *planner) expandSubqueries(stmt parser.Statement) error {
 func (p *planner) selectIndex(s *scanNode) (planNode, error) {
 	if s.desc == nil || s.filter == nil {
 		// No table or where-clause.
+		s.initOrdering()
```

Comment:
Oh, I remember now. `p.selectIndex` returns a `planNode`. Right now the `planNode` is always a `*scanNode`, but this will change fairly soon when we perform more interesting index lookups.